### ISSUE TITLE
Create multiple empty node groups by default

### ISSUE THREAD

**Issue by jrsdav** [streamnative/terraform-aws-cloud#84]
The module should be refactored to create multiple node groups using set(s) of AWS EC2 Instance Classes that correlate to a StreamNative cluster size.

The node groups should have a `desired_capacity` setting of `0`, allowing them to scale up/down according to what workloads are being scheduled by the StreamNative control plane.

Depending on regional availability:
- c6i.large, c6i.xlarge, c6i.2xlarge, c6i.4xlarge

Or:

- c5.large, c5.xlarge, c5.2xlarge, c5.4xlarge 

Other requirements:
- The name structure should match the naming pattern `"snc-*-pool*"`
  - Have some variation of the instance class in their name
  - The node group name should have a unique suffix

Nice to have:
- Include a taint that designates its instance class (micro, tiny, small, medium, large, xlarge, xxlarge). This will require some coordination with the control plane, so it should not be used until it is supported on workloads. Example value structure:

```
taints = [{
      key    = "reserveGroup"
      value  = "micro"
      effect = "NO_SCHEDULE"
    }]
```

**Comment by gengmao** [comment]
@jrsdav Inspired by [this comment](https://github.com/terraform-aws-modules/terraform-aws-eks/issues/1415#issuecomment-857952296), I found the `terraform-aws-eks` module actually supports creating launch template with instance types and don't specify instance types when creating node group via a `set_instance_types_on_lt` variable which is available in [`v17.4.0`](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/v17.4.0/modules/node_groups/node_groups.tf#L19). That will allow us to update instance types in the launch template via Terraform without impacting existing node groups, then trigger an Instance Refresh manually to rotate instances. 

The drawback of setting `set_instance_types_on_lt` is it breaks backward compatibility on existing clusters/node groups. Hence I created https://github.com/streamnative/terraform-aws-cloud/pull/85 to simplify updating instance types for existing clusters. 

So different strategies for different clusters.

**Comment by gengmao** [comment]
This was done with #91
